libvast/vast/system/index_common.hpp

```diff
@@ -0,0 +1,66 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include "vast/aliases.hpp"
+#include "vast/fwd.hpp"
+#include "vast/type.hpp"
+
+#include <tuple>
+#include <vector>
+
+namespace vast {
+
+/// A standalone field of an event type, used to address an index column.
+/// Example: { "zeek.conn.id.orig_h", address_type{} }
+struct fully_qualified_leaf_field {
+  std::string name; ///< The name of the field.
+  vast::type type;  ///< The type of the field.
+
+  vast::record_field to_record_field() const;
+
+  friend bool operator==(const fully_qualified_leaf_field& x,
+                         const fully_qualified_leaf_field& y);
+
+  friend bool operator<(const fully_qualified_leaf_field& x,
+                        const fully_qualified_leaf_field& y);
+
+  template <class Inspector>
+  friend auto inspect(Inspector& f, fully_qualified_leaf_field& x) {
+    return f(x.name, x.type);
+  }
+};
+
+struct slice_column {
```

Comment:
The type is now in its own file, but I did not change it's API yet. I think this should rather be done in a different PR, because it will also affect the `column_index`.